packager/packager.cc

```diff
@@ -773,13 +929,26 @@ class ImportTreeBuilder final {
             }
             node = child.get();
         }
-        node->source = {importedPackage.name.mangledName, loc, importType};
+
+        // If the node already has an import source, this is a conflicting import.
+        // See test/cli/package-import-conflicts/ for an example.
+        if (node->source.exists() && importType != ImportType::Self) {
+            addImportConflictError(ctx, loc, node->source.importLoc, exportFqn.parts);
+        }
+        node->source = {importedPackage.name.mangledName, loc, importType, isEnumeratedImport};
```

Comment:
I know this didn't originate in this PR, but using the name of the type here helps with readability. Alternatively you could assign each of the fields separately.
```suggestion
        node->source = ImportTree::Source{importedPackage.name.mangledName, loc, importType, isEnumeratedImport};
```